src/__tests__/widget_test.ts

```diff
@@ -326,6 +326,43 @@ widgets.forEach((widgetClass) => {
 })
 
 describe("ConnectWidget", () => {
+  describe("handleOAuthRedirect", () => {
+    test("should throw with a bad URL", async () => {
+      expect(() => {
+        const widget = new ConnectWidget({ url, container })
+
+        widget.handleOAuthRedirect("")
+      }).toThrow()
+    })
+
+    test("success URL", async () => {
+      const widget = new ConnectWidget({ url, container })
+
+      await waitFor(() => !!container?.getElementsByTagName("iframe")[0]?.src)
+
+      const iframeElement = container?.getElementsByTagName("iframe")[0]?.contentWindow
+
+      if (!iframeElement) {
+        throw new Error("Unable to find widget iframe")
+      }
+
+      const postMessageSpy = jest.spyOn(iframeElement, "postMessage")
+
+      widget.handleOAuthRedirect("com.testapp://oauth_complete?status=error&member_guid=MBR-123")
```

Comment:
Can we trigger this with a post message instead? That'll test the message listener as well.

Reply:
Along these lines, can we also add a test that checks that a post message that's not ours doesn't trigger a post message to the iframe?